File: src/protocol/MotorStatusResponse.cpp
#include "MotorStatusResponse.h"

#include "ResponseVisitor.h"

using namespace std;
using namespace communication;

MotorStatusResponse::MotorStatusResponse(MotorStatus type)
        : Response(10, ResponseType::MOTOR_STATUS),
          type_(type)
{}

MotorStatusResponse::~MotorStatusResponse()
{}

vector<uint8_t> MotorStatusResponse::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(responseType_));
    frame.push_back(static_cast<uint8_t>(type_));

    return frame;
}

string MotorStatusResponse::getName()
{
    return string("MotorStatusResponse");
}

void MotorStatusResponse::accept(ResponseVisitor &visitor)
{
    visitor.visit(*this);
}

MotorStatus MotorStatusResponse::getMotorStatus() const
{
    return type_;
}

void MotorStatusResponse::setMotorStatus(MotorStatus type)
{
    type = type_;
}

void MotorStatusResponse::initializeDataSize()
{
    uint16_t dataSize = sizeof(responseType_);
    dataSize += sizeof(type_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/protocol/MotorStatusResponse.java
package protocol;

import java.util.ArrayList;

public final class MotorStatusResponse extends Response
	{
		public MotorStatusResponse(MotorStatus type)
		{
			super(10, ResponseType.MOTOR_STATUS);
			this.type_ = type;
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(responseType_.getValue());
			frame.add(type_.getValue());

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("MotorStatusResponse");
		}

		@Override
		public void accept(ResponseVisitor visitor)
		{
			visitor.visit(this);
		}

		public MotorStatus getMotorStatus()
		{
			return this.type_;
		}
		public void setMotorStatus(MotorStatus type)
		{
			this.type_ = type;
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(2);
		}

		private MotorStatus type_;
	}